src/server/transaction.h

```diff
@@ -636,6 +647,8 @@ class Transaction {
     ShardId coordinator_index = 0;
   } stats_;
 
+  ConnectionContext* cntx_{nullptr};
+
```

Comment:
I'll invent something in the future to get rid of this 😆

Reply:
No need to wait for the future, I think transactions should not be aware of ConnectionContext. It's a design choice and it is possible to preserve it. Let's introduce a `on_track_cb` that will be passed to transaction if tracking is needed, i.e. if the condition `((cid->opt_mask() & CO::READONLY) && cid->IsTransactional() && info.ShouldTrackKeys())` holds. The callback will call `OpTrackKeys`. This way `ClientTracking::Track` will disappear at all